packages/table/test/table.test.ts

```diff
@@ -0,0 +1,1043 @@
+        const newValue = tableRow.value;
+
+        expect(newValue).to.not.equal(initialValue);
+    });
+
+    it('dispatches `visibilityChanged` events on Virtualized Table', async () => {
+        // console.log('NOT this test');
+        const visibilityChangedSpy = spy();
+        const virtualItems = makeItemsTwo(50);
+
+        const el = await fixture<Table>(html`
+            <sp-table
+                selects="multiple"
+                .selected=${['1', '47']}
+                style="height: 120px"
+                .items=${virtualItems}
+                .renderItem=${renderItem}
+                scroller?="true"
+                @visibilityChanged=${({
+                    target,
+                }: Event & { target: Table }) => {
+                    visibilityChangedSpy(target);
+                }}
+            >
+                <sp-table-head>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                </sp-table-head>
+            </sp-table>
+        `);
+        await nextFrame();
+        await nextFrame();
+        await nextFrame();
+
+        expect(el.selected).to.deep.equal(['1', '47']);
+
+        el.scrollToIndex(47);
+
+        await nextFrame();
+        await nextFrame();
+        await elementUpdated(el);
+
+        expect(visibilityChangedSpy.called).to.be.true;
+    });
+
+    it('selects all checkboxes in Virtualized Table when clicking the TableHeadCheckboxCell', async () => {
+        const virtualItems = makeItemsTwo(50);
+
+        const el = await fixture<Table>(html`
+            <sp-table
+                selects="multiple"
+                .selected=${['1', '47']}
+                style="height: 120px"
+                .items=${virtualItems}
+                .renderItem=${renderItem}
+                scroller?="true"
+            >
+                <sp-table-head>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                </sp-table-head>
+            </sp-table>
+        `);
+        await elementUpdated(el);
+
+        expect(el.selected).to.deep.equal(['1', '47']);
+
+        const tableHeadCheckboxCell = el.querySelector(
+            'sp-table-head sp-table-checkbox-cell'
+        ) as TableCheckboxCell;
+
+        tableHeadCheckboxCell.checkbox.click();
+
+        await elementUpdated(el);
+    });
+
+    it('validates `value` property to make sure it matches the values in `selected`', async () => {
+        const el = await fixture<Table>(
+            virtualizedCustomValue(virtualizedCustomValue.args as Properties)
+        );
+
+        expect(el.selected).to.deep.equal(['applied-47']);
```

Comment:
We should also apply an invalid `selected` value post render to ensure we cover this use case in both ways it can come up for a consumer.